be/src/storage/storage_engine.cpp

```diff
@@ -898,6 +900,17 @@ Status StorageEngine::_perform_base_compaction(DataDir* data_dir, std::pair<int3
     return Status::OK();
 }
 
+Status StorageEngine::_perform_pk_index_bg_compaction(DataDir* data_dir) {
+    TabletSharedPtr best_tablet = _tablet_manager->find_best_tablet_to_do_pk_index_bg_compaction(data_dir);
+    if (best_tablet == nullptr) {
+        return Status::NotFound("there are no suitable tablets");
+    }
+    if (best_tablet->updates() == nullptr) {
+        return Status::InternalError("not an updatable tablet");
+    }
+    return best_tablet->updates()->pk_index_bg_compaction();
```

Comment:
There may be multiple threads calling on the same dir, and pickup same tablet?
It seems you need a single thread to schedule compaction task and send picked up compaction tablet to threadpool?

Reply:
Done, using threadpool to handle this.